File: LeetCode/cpp/2500_Delete_Greatest_Value_in_Each_Row.cpp
class Solution {
public:
	int deleteGreatestValue(vector<vector<int>>& grid) {
		int answer = 0;

		for (int i = 0; i < grid.size(); i++) {
			sort(grid[i].begin(), grid[i].end(), greater<>());
		}

		for (int i = 0; i < grid[0].size(); i++) {
			int max_num = -1;

			for (int j = 0; j < grid.size(); j++) {
				max_num = max(max_num, grid[j][i]);
				grid[j][i] = -1;
			}

			answer += max_num;
		}

		return answer;
	}
};

File: LeetCode/java/2500_Delete_Greatest_Value_in_Each_Row.java
import java.util.Arrays;

class Solution {
    public int deleteGreatestValue(int[][] grid) {
        int answer = 0;

        for (int i = 0; i < grid.length; i++) {
            Arrays.sort(grid[i]);
        }

        for (int i = grid[0].length - 1; i >= 0; i--) {
            int maxNum = -1;

            for (int j = 0; j < grid.length; j++) {
                maxNum = Math.max(maxNum, grid[j][i]);
                grid[j][i] = -1;
            }

            answer += maxNum;
        }

        return answer;
    }
}